### ISSUE TITLE
Payment error - rescan required

### ISSUE THREAD

**Issue by tisek** [voisine/breadwallet#196]
I just did an payment and got an error telling me I should rescan the blockchain. I did a rescan and retried the payment. This time all went fine...

A short time later when I had a look in the transaction history, I noticed that I paid 2 times the same amount to the same address :-(. The target is BitPay so it won't be a problem getting the 2nd coins back. But the app should prevent such human errors.

1) The error message (I don't remember the exact text) was misleading. I thought that I had to try the payment again. This should be fixed.

2) If you try to pay the same amount again to the same address when the payment has not at least 6 confirmations, this should trigger a warning like "Are you sure that you want to pay the same amount to this address again? Maybe the target is just waiting for an confirmation before processing your payment."

**Comment by voisine** [comment]
Thanks for reporting. Adding a warning when sending the same amount twice in a row to the same address is in a "todo" item in the code somewhere.

There are also several improvements in the code since the last app store update that will improve the detection of when the network rejects an unconfirmed transaction, so you won't get that rescan warning unless the network really is rejecting your spends.

**Comment by dabura667** [comment]
So "rescan required" meant "rescan the blockchain using the option in settings" but the user saw it as "rescan the QR code of the payment you just sent." and accidentally sent.

"rescan required" is a misleading error, if it exists in breadwallet we should change to "Please perform "rescan blockchain" from the settings." etc.

**Comment by voisine** [comment]
Excellent point @dabura667. I will change that to remove any ambiguity.

**Comment by kristovatlas** [comment]
@voisine breadwallet is one of the only wallets that warns about sending to the same address twice. nice!

unfortunately, it only seems to provide this warning when the offending bitcoin address is copied to the clipboard. If you, for example, scan the address in via QR code, no warning. Is it possible to show this warning more consistently regardless of pasteboard state?

Also, the next level up would be to warn about sending to addresses that have a tx history (that you may not have sent to personally in the past). This would require a balance lookup query for the address in question, of course.

**Comment by voisine** [comment]
@kristovatlas yes, that was intentional, targeted at the case where a user copied an address to the clipboard from their transaction history. When scanning a QR or tapping a URL we felt like it was more important to just do what the user expects rather than annoying them with a warning. However as more wallets implement HD and merchants continue to move to payment processor services over static addresses, making the warning occur for every address reuse starts to make more sense.

**Comment by kristovatlas** [comment]
It's definitely a tricky balance between potentially inconveniencing users with unwanted information and steering them in the right direction when they need it. I'm no UX designer, but would prefer the warning to be on always by default and easily disable-able by the user. The consistent application of the warning is a criteria for OBPP's wallet ratings.

**Comment by Logicwax** [comment]
Address re-use, I think, is a problem that is outside of the scope of breadwallet.   It is the problem of the payee.  The endpoint outside of breadwallet.  I don't really see why breadwallet should be responsible for what people do outside of breadwallet.
